test/rules/order/ordering.js

```diff
@@ -185,4 +289,142 @@ describe('Linter - ordering', () => {
+    const code = `
+      // SPDX-License-Identifier: MIT
+      pragma solidity ^0.8.0;
+      import "@openzeppelin/contracts/ownership/Ownable.sol";
+      function freeFunction() pure returns (uint256) {
+        return 1;
+      }     
+      error Unauthorized();
+      contract OneNiceContract {}
+    `
+    const report = linter.processStr(code, {
+      rules: { ordering: 'error' },
+    })
+
+    assert.equal(report.errorCount, 1)
+    assert.ok(
+      report.messages[0].message.includes(
+        'Function order is incorrect, custom error definition can not go after free function definition'
+      )
+    )
+  })
+
+  it('should not raise error when free function is defined in correct order', () => {
+    const code = `
+      // SPDX-License-Identifier: MIT
+      pragma solidity ^0.8.0;
+      import "@openzeppelin/contracts/ownership/Ownable.sol";
+      error Unauthorized();
+      function freeFunction() pure returns (uint256) {
+        return 1;
+      }     
+      contract OneNiceContract {}
+    `
+    const report = linter.processStr(code, {
+      rules: { ordering: 'error' },
+    })
+
+    assert.equal(report.errorCount, 0)
+  })
+
+  it('should raise error when file level constant is defined after free function', () => {
+    const code = `
+      // SPDX-License-Identifier: MIT
+      pragma solidity ^0.8.0;
+      import "@openzeppelin/contracts/ownership/Ownable.sol";
+      function freeFunction() pure returns (uint256) {
+        return 1;
+      }     
+      uint256 constant oneNiceConstant = 1;
+      contract OneNiceContract {}
+    `
+    const report = linter.processStr(code, {
+      rules: { ordering: 'error' },
+    })
+
+    assert.equal(report.errorCount, 1)
+    assert.ok(
+      report.messages[0].message.includes(
+        'Function order is incorrect, file level constant can not go after free function definition'
+      )
+    )
+  })
+
+  it('should not raise error when file level constant is defined in correct order', () => {
+    const code = `
+      // SPDX-License-Identifier: MIT
+      pragma solidity ^0.8.0;
+      import "@openzeppelin/contracts/ownership/Ownable.sol";
+      uint256 constant oneNiceConstant = 1;
+      function freeFunction() pure returns (uint256) {
+        return 1;
+      }     
+      contract OneNiceContract {}
+    `
+    const report = linter.processStr(code, {
+      rules: { ordering: 'error' },
+    })
+
+    assert.equal(report.errorCount, 0)
+  })
+
+  it('should not raise error when all top level code is well placed', () => {
+    const code = `
+      // SPDX-License-Identifier: MIT
+      pragma solidity ^0.8.0;
+      import "@openzeppelin/contracts/ownership/Ownable.sol";
+      error Unauthorized();
+      enum MyEnum { A, B }
+      struct OneNiceStruct { uint256 a; uint256 b; }            
+      uint256 constant oneNiceConstant = 1;
+      function freeFunction() pure returns (uint256) {
+        return 1;
+      }     
```

Comment:
@frangio sorry for the ping, but do you have an opinion on this order for file-level nodes? The new supported elements are custom errors, constants and functions.

I feel like I would swap constants and errors here, so that the order is:
- Constants
- Enum/Structs
- Custom errors
- Functions
- Contracts

Bikeshedding though.

Reply:
No opinion here to be honest.

Reply:
@fvictorio I feel the order you proposed is a better one
I changed the weight so it can match what you suggested
Please approve when you have time